src/components/dragons.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchDragon from './API/apiDragons';
+import { reserveDragon, cancelDragon } from '../features/dragonSlice';
+
+function Dragon({
+  id, name, type, image, reserved,
+}) {
+  const dispatch = useDispatch();
+
+  const handleReserve = () => {
+    dispatch(reserveDragon({ id }));
+  };
+
+  const handleCancel = () => {
+    dispatch(cancelDragon({ id }));
+  };
+
+  return (
+    <div className="dragons_container">
+      <h1>{name}</h1>
+      <p>{type}</p>
+      <img src={image} alt={name} width="300" height="300" />
+      {reserved ? (
+        <button onClick={handleCancel} type="button">
+          Cancel booking
+        </button>
+      ) : (
+        <button onClick={handleReserve} type="button">
+          Reserve dragon
+        </button>
+      )}
+    </div>
+  );
+}
+function ReserveDragons() {
+  const dragons = useSelector((state) => state.dragonReducer.dragonStore);
+  const filteredDragons = dragons.filter((dragon) => dragon.reserved);
+
+  return (
+    <div>
+
+      {filteredDragons.map((dragon) => (
+        <div key={dragon.id}>{dragon.name}</div>
+      ))}
+    </div>
+  );
+}
+
+Dragon.propTypes = {
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  image: PropTypes.string.isRequired,
+  reserved: PropTypes.bool.isRequired,
+};
+
+function Dragons() {
+  const dragonsArr = useSelector((state) => state.dragonReducer.dragonStore);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchDragon());
+  }, [dispatch]);
```

Comment:
- [ ] You really did an amazing job. unlike the rockets, every time I make dragon reservations, navigate and come back to the dragon page, the state is re-initialized and all my reservations are wiped out please kindly fix it